hooks/configure

```diff
@@ -102,7 +109,50 @@ END {
         printf("%s=%s\n",ENVIRON["key"],ENVIRON["value"])
 }
 EOF
-            mv "${path}.tmp" "$path"
+    mv "${tmp_path}" "$path"
+}
+
+update_pi_config_value() {
+    path="$1"
+    key="$2"
+    value="$3"
+
+    if [ -z "$value" ]; then
+        # empty - unset an option
+        if grep -q "^[ \t]*${key}=" "$path"; then
+            tmp_path="${path}.tmp"
+
+            sed "/^[ \t#]*${key}=/ s/^#*/#/" "$path" > "${tmp_path}"
+            mv "${tmp_path}" "$path"
+        fi
+    else
+        if ! grep -q "^[ \t]*${key}=${value}$" "$path"; then
+            # non-empty - set an option
+            key_value_replace_or_append "$key" "$value" "$path"
+        fi
+    fi
+}
+
+switch_proxy() {
+    path="$1"
+    proxy="${2}_proxy"
+    value="$3"
+
+    if [ -z "$value" ]; then
+        if grep -q "^${proxy}=.*" "$path";then
+            tmp_path="${path}.tmp"
+            # not writable, most likely due to a read-only directory, e.g. when
+            # dealing with /etc/envirionment on core
+            if ! is_writable "$tmp_path"; then
+                tmp_path="$(mktemp)"
+            fi
+
+            sed  "/^${proxy}=.*/d" "$path" > "${tmp_path}"
+            mv "${tmp_path}" "${path}"
+        fi
+    else
+        if ! grep -q "^${proxy}=${value}" "$path";then
```

Comment:
Some inconsistent spacing before then.

I'm not sure if the grep is right in value of various possible (valid) values of the ${value}. Here it will be treated as a regular expression.